modules/base_client.go

```diff
@@ -336,6 +335,58 @@ func (base *baseClient) ValidateTxSize(txSize int, msgs []sdk.Msg) sdk.Error {
 	return nil
 }
 
+func initConfig(cdc sdk.Codec, cfg *sdk.ClientConfig){
+	if len(cfg.NodeURI) == 0 {
+		panic(fmt.Errorf("nodeURI is required"))
+	}
+
+	if len(cfg.Network) == 0 {
+		cfg.Network = sdk.Mainnet
+	}
+
+	if len(cfg.ChainID) == 0 {
+		panic(fmt.Errorf("chainID is required"))
+	}
+
+	if cfg.Gas == 0 {
+		cfg.Gas = 20000
+	}
+
+	if cfg.Fee == nil || cfg.Fee.Empty() {
+		panic(fmt.Errorf("fee is required"))
+	}
+
+	if cfg.KeyDAO == nil {
+		if len(cfg.DBRootDir) == 0 {
+			panic(fmt.Errorf("DBRootDir is required when use default keyDao"))
```

Comment:
Don't we have a default root dir?

Reply:
The storage path of the db file is very important, the default path may cause users to accidentally lose data files